### PR TITLE
Add created_by and updated_by fields to base model

### PR DESCRIPTION
**Pull request by AnkurPrabhu**
## Proposed Changes

- Added 2 new fields in base model created_by and updated_by 
- created migrations for the same
- updated the create and update function of viewsets to add values to created_by and updated_by fields

### Associated Issue
- Link to issue #1697  

## Merge Checklist

- [ ] Tests added/fixed
- [ ] Update docs in `/docs`
- [ ] Linting Complete
- [ ] Any other necessary step

_*Only PR's with test cases included and passing lint and test pipelines will be reviewed*_

@coronasafe/care-backend-maintainers @coronasafe/care-backend-admins

### CODE REVIEW

**Comment by sainak** [comment]
initial changes look good, you can proceed with adding created_by updated_by to other models as well

**Comment by AnkurPrabhu** [comment]
> initial changes look good, you can proceed with adding created_by updated_by to other models as well

Hey i made the changes at all the places but i realised that perform_update does not need to update created_by field while saving so i have done all the changes will test it locally and then push it for review

**Comment by AnkurPrabhu** [comment]
@sainak i have updated the pr, please take a look and let me know if there are any changes i have tested for some of the models it works fine for me

**Comment by sainak** [comment]
> @sainak i have updated the pr, please take a look and let me know if there are any changes i have tested for some of the models it works fine for me

hey @AnkurPrabhu, thanks for these changes, currently focused on some high-priority issues, I'll surely review this pr once I'm done with those

**Review by sainak — CHANGES_REQUESTED** [review]

File: care/facility/api/viewsets/patient_otp.py [hunk 1]
```diff
@@ -53,3 +53,6 @@ def login(self, request):
         token["phone_number"] = phone_number
 
         return Response({"access": str(token)})
+
+    def perform_create(self, serializer):
+        serializer.save(updated_by=self.request.user)
```
[the remark this answers is not in the document]
Comment (AnkurPrabhu) [reply care/facility/api/viewsets/patient_otp.py:58]: so remove it form here ?

File: care/facility/api/viewsets/mixins/access.py [hunk 2]
```diff
@@ -44,6 +44,10 @@ def perform_create(self, serializer):
         kwargs = {}
         if hasattr(model(), "created_by"):
             kwargs["created_by"] = self.request.user
+
+        if hasattr(model(), "updated_by"):
+            kwargs["updated_by"] = self.request.user
+
         serializer.save(**kwargs)
```
[the remark this answers is not in the document]
Comment (AnkurPrabhu) [reply care/facility/api/viewsets/mixins/access.py:57]: cool wil do

**Comment by AnkurPrabhu** [comment]
@sainak  these are the only changes ?

**Comment by AnkurPrabhu** [comment]
@sainak  any other changes required for this or only these 2

**Comment by sainak** [comment]
> @sainak any other changes required for this or only these 2

yes
make the changes and get it tested by @nihal467

**Comment by AnkurPrabhu** [comment]
@sainak have resolved your comments. @nihal467 please take a look, I have tested it for some endpoints in my local

File: care/facility/api/viewsets/mixins/access.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sainak) [reply care/facility/api/viewsets/mixins/access.py:57]: @AnkurPrabhu this was not resolved in the last commit

**Comment by AnkurPrabhu** [comment]
@nihal467  just a reminder to take a look, please take a look when you are available

File: care/facility/api/viewsets/mixins/access.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AnkurPrabhu) [reply care/facility/api/viewsets/mixins/access.py:57]: i have added perform update method right in commit https://github.com/coronasafe/care/pull/1940/commits/8e63db6df36b8a036ec99ac1f3483d83eeafb8e1

**Comment by AnkurPrabhu** [comment]
@sainak @nihal467  there were some merge conflicts due to recent db changes made by @rithviknishad I have now rebased it but can we get this tested ?
as the changes are in a lot of models and can conflict frequently with other branches

**Comment by nihal467** [comment]
@AnkurPrabhu 
image

the backend is not loading properly in docker, just check your PR is working properly , there are migration missing in it

**Comment by AnkurPrabhu** [comment]
> @AnkurPrabhu image
> 
> the backend is not loading properly in docker, just check your PR is working properly , there are migration missing in it


was working fine for me will take a look

**Comment by AnkurPrabhu** [comment]
@nihal467 there was a pr which was related to db change which was conflicting with my change deleted those migrations and created new ones.

**Comment by AnkurPrabhu** [comment]
@sainak @nihal467 the docker for me now shows update_by field already present is table ambulance but its not.
also it says "facility_ambulance" so how are they related i was not able to find this, the relation seems like the issue here 
 
Screenshot 2024-04-10 at 11 45 55 PM

**Comment by nihal467** [comment]
image

it's still throwing error, fix it

**Pull request closed without merging**